File: baseballserver/apps/media/gallery/serializers/album.py
from rest_framework import serializers

from ..models import Album


class AlbumSerializer(serializers.ModelSerializer):
    cover_image_url = serializers.SerializerMethodField()
    num_images = serializers.IntegerField(read_only=True, source="images.count")
    num_videos = serializers.IntegerField(read_only=True, source="videos.count")

    class Meta:
        model = Album
        fields = [
            "id",
            "title",
            "members_only",
            "color",
            "cover_image_url",
            "num_images",
            "num_videos",
        ]

    def get_cover_image_url(self, obj):
        ## random image from the album.
        images = obj.images.all()
        if images.exists():
            image = images.order_by("?").first()
            return image.image.url
        return None


class AlbumSimpleSerializer(serializers.ModelSerializer):
    class Meta:
        model = Album
        fields = ["id", "title", "members_only", "color"]
        read_only_fields = ["id"]

File: snubaseball/src/05.entities/albums/models/album.ts
export type AlbumType = {
  id: number;
  title: string;
  color: string;
  cover_image_url: string;
  num_images: number;
  num_videos: number;
};

export const sampleAlbums: AlbumType[] = [
  {
    id: 1,
    title: "2025년 시즌",
    color: "#FF5733",
    cover_image_url: "/images/sample_album_1.jpg",
    num_images: 120,
    num_videos: 5,
  },
  {
    id: 2,
    title: "2024년 시즌",
    color: "#33C1FF",
    cover_image_url: "",
    num_images: 95,
    num_videos: 3,
  },
];
